app/controllers/api/v2/accredited_provider_training_providers_controller.rb

```diff
@@ -6,7 +6,17 @@ class AccreditedProviderTrainingProvidersController < API::V2::ApplicationContro
 
       def index
         authorize @provider, :can_list_training_providers?
-        providers = @provider.training_providers
+        providers = if params[:filter]
+                      course_scope = Course.where(provider: training_providers)
+
+                      eligible_training_provider_ids = CourseSearchService
+                                                         .call(filter: params[:filter], course_scope: course_scope)
+                                                         .pluck(:provider_id)
+
+                      training_providers.where(id: eligible_training_provider_ids)
+                    else
+                      training_providers
+                    end
 
         render jsonapi: providers, include: params[:include]
```

Comment:
do we need to consider pagination in the api? if not then not, then theres no issue. if yes then we want the result set to be ordered so we can paginate through them

Reply:
@AgaDufrat - do you need pagination for your ticket? If need I'd be inclined to deploy this as is and do pagination as a separate ticket (if req'd)

Reply:
My ticket doesn't say anything about the. The highest possible number of training providers we can have on the pages that use this is 36. More of a product question.